File: macos/Sources/FullscreenPolicy.swift
import Foundation

enum FullscreenPolicy {
    static func shouldHide(settings: AppSettings, fullscreen: Bool, capturing: Bool) -> Bool {
        (settings.hideInFullscreen && fullscreen) || (settings.hideDuringCapture && capturing)
    }

    static func coversScreen(
        width: Double,
        height: Double,
        screenWidth: Double,
        screenHeight: Double,
        scale: Double
    ) -> Bool {
        func near(_ a: Double, _ b: Double) -> Bool { abs(a - b) < 4 }
        return (near(width, screenWidth) && near(height, screenHeight))
            || (near(width, screenWidth * scale) && near(height, screenHeight * scale))
    }

    static func isCaptureOwner(_ name: String) -> Bool {
        name.lowercased().contains("screencapture")
    }
}

File: windows/src/MonitorSuhu.Core/Services/FullscreenPolicy.cs
using MonitorSuhu.Core.Models;

namespace MonitorSuhu.Core.Services;

public static class FullscreenPolicy
{
    public static bool ShouldHide(AppSettings settings, bool fullscreen, bool capturing) =>
        (settings.HideInFullscreen && fullscreen) || (settings.HideDuringCapture && capturing);

    public static bool RectCoversMonitor(
        int left, int top, int right, int bottom,
        int monitorLeft, int monitorTop, int monitorRight, int monitorBottom,
        int tolerance = 2) =>
        Math.Abs(left - monitorLeft) <= tolerance
        && Math.Abs(top - monitorTop) <= tolerance
        && Math.Abs(right - monitorRight) <= tolerance
        && Math.Abs(bottom - monitorBottom) <= tolerance;

    public static bool CoversScreen(double width, double height, double screenWidth, double screenHeight, double scale)
    {
        static bool Near(double a, double b) => Math.Abs(a - b) < 4;
        return (Near(width, screenWidth) && Near(height, screenHeight))
               || (Near(width, screenWidth * scale) && Near(height, screenHeight * scale));
    }

    public static bool IsCaptureOwner(string? name)
    {
        if (string.IsNullOrEmpty(name)) return false;
        return name.Contains("screencapture", StringComparison.OrdinalIgnoreCase);
    }
}
